invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/ShareOptions/AccessLinks/AccessDropdown.js

```diff
@@ -28,16 +27,19 @@ export class AccessDropdown extends Component {
   }
 
   handleUpdate = async (permission) => {
-    const { record, result } = this.props;
+    const { updateEndpoint } = this.props;
     const data = { permission: permission };
     this.setState({ loading: true, actionSuccess: false });
-    this.cancellableAction = withCancel(
-      http.patch(`${record.links.access_links}/${result.id}`, data)
-    );
+    this.cancellableAction = withCancel(http.patch(updateEndpoint, data));
 
     try {
       await this.cancellableAction.promise;
-      this.setState({ loading: false, actionSuccess: true, error: undefined });
+      // prolong loading for better understanding that the action was completed
+      setTimeout(() => {
+        this.setState({ loading: false }, () => {
+          this.setState({ actionSuccess: true, error: undefined });
+        });
+      }, 500);
```

Comment:
This will make the UI to always wait for 500 ms after the end of the http request, isn't it?
What I meant during the demo was to have a minimal duration:
```javascript
// somewhere above as const
MIN_LOADING_DURATION = 200; // in milliseconds

const onSuccess = () =>  this.setState({ loading: false, actionSuccess: true, error: undefined });
try {
      const requestStartTime = Date.now();
      await this.cancellableAction.promise;

      const requestEndTime = Date.now();
      const requestDuration = requestEndTime - requestStartTime;

      if (requestDuration < MIN_LOADING_DURATION) {
          setTimeout(() => onSuccess(), MIN_LOADING_DURATION - requestDuration);
      } else {
          onSuccess();
      }

```